File: dotnet/base/Mcma.Client/Resources/ResourceManagerProvider.cs
using System;
using System.Net.Http;

namespace Mcma.Client
{
    public class ResourceManagerProvider : IResourceManagerProvider
    {
        public ResourceManagerProvider(IAuthProvider authProvider = null, ResourceManagerConfig defaultConfig = null)
        {
            AuthProvider = authProvider;
            DefaultConfig = defaultConfig;
        }

        private IAuthProvider AuthProvider { get; }

        private ResourceManagerConfig DefaultConfig { get; }

        public IResourceManager Get(HttpClient httpClient, ResourceManagerConfig config = null)
            => new ResourceManager(httpClient, ConfigOrDefault(config), AuthProvider);

        public IResourceManager Get(ResourceManagerConfig config = null)
            => new ResourceManager(ConfigOrDefault(config), AuthProvider);

        private ResourceManagerConfig ConfigOrDefault(ResourceManagerConfig config)
            => (config ?? DefaultConfig) ?? throw new McmaException("Config for resource manager not provided, and there is no default config available");
    }
} 

File: base/client/lib/resources/resource-manager-provider.ts
import { ConfigVariables, McmaException } from "@mcma/core";

import { AuthProvider } from "../auth";
import { ResourceManagerConfig } from "./resource-manager-config";
import { ResourceManager } from "./resource-manager";
import { getResourceManagerConfig } from "./config-variables-ext";

export class ResourceManagerProvider {
    constructor(private authProvider: AuthProvider, private defaultConfig?: ResourceManagerConfig) {
    }

    get(config?: ResourceManagerConfig | ConfigVariables) {
        config = config || this.defaultConfig;
        if (!config) {
            config = ConfigVariables.getInstance();
        }
        if (config instanceof ConfigVariables) {
            config = getResourceManagerConfig(config);
        }
        if (!config) {
            throw new McmaException("Config for resource manager not provided, and there is no default config available");
        }
        return new ResourceManager(config, this.authProvider);
    }
}
